### ISSUE TITLE
Catching "CURL Error: Transferred a partial file"?

### ISSUE THREAD

**Issue by jypeter** [CDAT/cdms#102]
### The bug part

@dnadeau4 I used the script at the end of #98 several times yesterday, and in one case (see below), a **CURL Error: Transferred a partial file** error message was displayed, but it was not caught/reported by cdms2. That is, the script seemed to end correctly but it's possible that the downloaded data was corrupted...

```
(cdatm14) bash-4.1$ python -i ~/CDAT/Progs/Devel/carmen/download_NERSC_data.py 
Processing year 1851 Tue Feb 28 17:21:58 2017
  Opening http://portal.nersc.gov/pydap/20C_Reanalysis_version2c_ensemble/first_guess.derived/t2m/t2m_1851.daily.nc
Processing year 1852 Tue Feb 28 17:21:59 2017
  Opening http://portal.nersc.gov/pydap/20C_Reanalysis_version2c_ensemble/first_guess.derived/t2m/t2m_1852.daily.nc
Processing year 1853 Tue Feb 28 17:22:00 2017
  Opening http://portal.nersc.gov/pydap/20C_Reanalysis_version2c_ensemble/first_guess.derived/t2m/t2m_1853.daily.nc
CURL Error: Transferred a partial file
>>> 
```

This may be hard to reproduce (need to overload the DAP server?), because it's probably due to a problem on the DAP server side. Still, it may be useful to check if all the reported errors are caught, since it is already difficult to monitor what really happens when you are talking to a DAP server.

### A related enhancement part
In the case above, I was transferring small chunks and I got 1 file/second. But when I started working with bigger chunks, nothing was happening during a long time and it was hard to determine if it was just normally slow or if the script or transfer was broken. Does cdms2 requests the big chunks specified by the users, or can it automatically break them into smaller chunks (possibly more efficient?) and display an optional progress bar

**Comment by dnadeau4** [comment]
I will look into it...

**Comment by jypeter** [comment]
I have gotten this CURL error several times today, even when downloading small chunks of files. Contrary to what I thought, the data did not seem corrupted so maybe there is some mechanism that already downloads the data again when this error takes place

**Comment by dnadeau4** [comment]
@jypeter make sure you delete the ~/.dodsrc file.  OpenDAP files get in trouble because of this.  CDMS got a segmentation fault using libnetcdf with an erroneous ~/.dodsrc.

**Comment by dnadeau4** [comment]
If I round the limits of the automatically generated bounds to 4 decimals (1e-4) it works in this case....
```python 
retbnds[0,0] = round(retbnds[0,0],4)
retbnds[-1,1] = round(retbnds[-1,1],4)
```

**Comment by jypeter** [comment]
I have a very old config file from 2003. See below. The idea at the time was to avoid having a cache in a hidden sub-directory of my backed-up home directory.

Any idea of what I should have in that file in order to do that?

```
 bash-4.1$ cat ~/.dodsrc
# DODS client configuation file. See the DODS
# users guide for information.
USE_CACHE=0
MAX_CACHE_SIZE=20
MAX_CACHED_OBJ=5
IGNORE_EXPIRES=0
CACHE_ROOT=/home/scratch01/jypeter/dods_cache/
DEFAULT_EXPIRES=86400
ALWAYS_VALIDATE=0
# PROXY_SERVER=<protocol>,<host url>
# PROXY_FOR=<regex>,<proxy host url>,<flags>
# NO_PROXY_FOR=<protocol>,<host>
```

**Comment by doutriaux1** [comment]
@jypeter I don't think this parameters affect .dodsrc, but try removing first. If it helps please put it back with all lines commented and uncomment only what you really need. I think the CACHE shouldn't matter.

**Comment by doutriaux1** [comment]
@dnadeau4 we should report the seg fault issue to netcdf

**Comment by dnadeau4** [comment]
@jypeter 
I just tried to read the first 20 arrays([0-19], Lev,lat,lon) in this file and everything is working fine with the latest `CDMS`.

Can you try this script?
```python
import cdms2
f=cdms2.open("http://portal.nersc.gov/pydap/20C_Reanalysis_version2c_ensemble/first_guess.derived/t2m/t2m_1851.daily.nc")
d=f['t2m']
d.getTime()
for i in range(20):  d[i,:,:,:]


```
